lib/modules/hosts/youtube.js

```diff
@@ -5,7 +5,7 @@ export default {
 	name: 'youtube',
 	attribution: false,
 	domains: ['youtube.com', 'youtu.be'],
-	detect: ({ href }) => (/^https?:\/\/(?:youtu\.be|(?:www\.|m\.)?youtube\.com)\/?(?:watch|embed)?(?:.*?v=|v\/|\/)([\w-]+)/i).exec(href),
+	detect: ({ href }) => (/^https?:\/\/(?:www\.|m\.)?(?:youtu\.be\/|youtube\.com\/?(?:watch|embed)(?:.*?v=|v\/|\/))([\w-]+)/i).exec(href),
```

Comment:
Honestly this Regex looks to be getting a bit bloated to the point where it's becoming difficult to read. Maybe it's time to drop the Regex here and use [`URL`](https://developer.mozilla.org/en-US/docs/Web/API/URL) or something like that instead? (Does `URL` work in Edge?)